tests/test_jwt.py

```diff
@@ -187,3 +194,143 @@ def test_roundtrip_explicit_key_id(token_factory):
+
+    @pytest.fixture(autouse=True)
+    def credentials(self, signer):
+        self.credentials = jwt.Credentials(
+            signer, self.service_account_email)
+
+    def test_from_service_account_info(self):
+        with open(SERVICE_ACCOUNT_JSON_FILE, 'r') as fh:
+            info = json.load(fh)
+
+        credentials = jwt.Credentials.from_service_account_info(info)
+
+        assert credentials._signer.key_id == info['private_key_id']
+        assert credentials._issuer == info['client_email']
+        assert credentials._subject == info['client_email']
+
+    def test_from_service_account_info_args(self):
+        info = dict(SERVICE_ACCOUNT_INFO)
+
+        credentials = jwt.Credentials.from_service_account_info(
+            info, subject='subject', audience='audience',
+            additional_claims={'meta': 'data'})
+
+        assert credentials._signer.key_id == info['private_key_id']
+        assert credentials._issuer == info['client_email']
+        assert credentials._subject == 'subject'
+        assert credentials._audience == 'audience'
+        assert credentials._additional_claims['meta'] == 'data'
+
+    def test_from_service_account_bad_key(self):
+        info = dict(SERVICE_ACCOUNT_INFO)
+        info['private_key'] = 'garbage'
+
+        with pytest.raises(ValueError) as excinfo:
+            jwt.Credentials.from_service_account_info(info)
+
+        assert excinfo.match(r'No key could be detected')
+
+    def test_from_service_account_bad_format(self):
+        with pytest.raises(ValueError):
+            jwt.Credentials.from_service_account_info({})
+
+    def test_from_service_account_file(self):
+        info = dict(SERVICE_ACCOUNT_INFO)
+
+        credentials = jwt.Credentials.from_service_account_file(
+            SERVICE_ACCOUNT_JSON_FILE)
+
+        assert credentials._signer.key_id == info['private_key_id']
+        assert credentials._issuer == info['client_email']
+        assert credentials._subject == info['client_email']
+
+    def test_from_service_account_file_args(self):
+        info = dict(SERVICE_ACCOUNT_INFO)
+
+        credentials = jwt.Credentials.from_service_account_file(
+            SERVICE_ACCOUNT_JSON_FILE, subject='subject', audience='audience',
+            additional_claims={'meta': 'data'})
+
+        assert credentials._signer.key_id == info['private_key_id']
+        assert credentials._issuer == info['client_email']
+        assert credentials._subject == 'subject'
+        assert credentials._audience == 'audience'
+        assert credentials._additional_claims['meta'] == 'data'
+
+    def test_default_state(self):
+        assert not self.credentials.valid
+        # Expiration hasn't been set yet
+        assert not self.credentials.expired
+
+    def test_sign_bytes(self):
+        to_sign = b'123'
+        signature = self.credentials.sign_bytes(to_sign)
+        crypt.verify_signature(to_sign, signature, PUBLIC_CERT_BYTES)
```

Comment:
No actual asserts here. How could you test something, other than a successful exit?